### PR TITLE
adapt gpcloud installcheck for mulitple platforms

### PR DESCRIPTION
**Pull request by goutamtadi1**
- add regression test for gpcloud on ubuntu
- create pattern for reusing task across platforms
- Remove configure step since it is not necessary
- Remove unused files

### CODE REVIEW

File: concourse/pipelines/pipeline.yml [hunk 1]
```diff
@@ -1178,6 +1178,38 @@ jobs:
...
+    - task: regression_tests_gpcloud_ubuntu
+      input_mapping:
+        bin_gpdb: compiled_bits_gpdb_with_orca_conan_ubuntu16
+      file: gpdb_src/concourse/tasks/regression_tests_gpcloud.yml
+      image: ubuntu-gpdb-dev-16
+      params:
+        gpcloud_access_key_id: {{gpcloud-access-key-id}}
+        gpcloud_secret_access_key: {{gpcloud-secret-access-key}}
+        TARGET_OS: ubuntu
+
```
Comment (kmacoskey) [inline concourse/pipelines/pipeline.yml]: Lacking some context: Would this ever need to include sles11/12 ?

File: concourse/pipelines/pipeline.yml [hunk 2]
```diff
@@ -1178,6 +1178,38 @@ jobs:
       BLDWRAP_POSTGRES_CONF_ADDONS: "fsync=off"
       TEST_OS: centos
 
+- name: icw_extensions_gpcloud
```
Comment (kmacoskey) [inline concourse/pipelines/pipeline.yml]: Lacking some context: Why is the test being moved out of the UD team group and into ICW? Is this just because ubuntu is being added?

File: concourse/pipelines/pipeline.yml [hunk 3]
```diff
@@ -1178,6 +1178,38 @@ jobs:
...
+      file: gpdb_src/concourse/tasks/regression_tests_gpcloud.yml
+      image: centos-gpdb-dev-6
+      params:
+        gpcloud_access_key_id: {{gpcloud-access-key-id}}
+        gpcloud_secret_access_key: {{gpcloud-secret-access-key}}
+        TARGET_OS: centos
+    - task: regression_tests_gpcloud_ubuntu
+      input_mapping:
+        bin_gpdb: compiled_bits_gpdb_with_orca_conan_ubuntu16
+      file: gpdb_src/concourse/tasks/regression_tests_gpcloud.yml
```
Comment (kmacoskey) [inline concourse/pipelines/pipeline.yml]: This yaml file still has this defined in it:
```
image_resource:
  type: docker-image
  source:
    repository: pivotaldata/centos-gpdb-dev
    tag: '6-gcc6.2-llvm3.7'
```
So setting `image: ubuntu-gpdb-dev-16` is conflicting

File: concourse/scripts/build_gpdb.py [hunk 4]
```diff
@@ -10,6 +11,7 @@
 ORCA_CODEGEN_DEFAULT_MODE = "orca_codegen"
 ORCA_MODE = 'orca'
 PLANNER_MODE = 'planner'
+INSTALL_DIR = "/usr/local/gpdb"
```
Comment (kmacoskey) [inline concourse/scripts/build_gpdb.py]: Why are we using `/usr/local/gpdb` in place of the standard `/usr/local/greenplum-db`?

File: concourse/scripts/builds/GpBuild.py [hunk 5]
```diff
@@ -1,27 +1,29 @@
...
         self.mode = 'on' if mode == 'orca' else 'off'
 
     def configure(self):
-        return subprocess.call(["./configure",
-                                "--enable-mapreduce",
-                                "--with-gssapi",
-                                "--with-perl",
-                                "--with-libxml",
-                                "--with-python",
-                                "--disable-gpcloud",
```
Comment (kmacoskey) [inline concourse/scripts/builds/GpBuild.py]: Probably lacking context: Why was this here in the first place to `--disable-gpcloud`. It is now not included in the new `config_options`.

File: concourse/pipelines/pipeline.yml (hunk 2 above)
Comment (goutamtadi1) [reply concourse/pipelines/pipeline.yml]: This test is basically running `install-check` on gpcloud. Considering all install checks are relevant to be done in IC group we moved this out of UD.

File: concourse/pipelines/pipeline.yml (hunk 1 above)
Comment (goutamtadi1) [reply concourse/pipelines/pipeline.yml]: Currently there is only testing CentOS and extending support to Ubuntu. We are unsure why sles is not included.

File: concourse/scripts/build_gpdb.py (hunk 4 above)
Comment (goutamtadi1) [reply concourse/scripts/build_gpdb.py]: This has been a practice for us since the Ubuntu work initiated and gporca uses `/usr/local/gpdb` and in the future planning to abstract this install dir path. This abstraction is only available if using python scripts to install gpdb

File: concourse/scripts/builds/GpBuild.py (hunk 5 above)
Comment (goutamtadi1) [reply concourse/scripts/builds/GpBuild.py]: Since this PR is to bring gpcloud tests to IC Group, we don't want to disable gpcloud. This avoids recompiling gpcloud in a separate job as a whole.

File: concourse/scripts/regression_tests_gpcloud.bash [hunk 6]
```diff
@@ -2,64 +2,72 @@
...
-	source /opt/gcc_env.sh
-	source /usr/local/greenplum-db-devel/greenplum_path.sh
+source \$INSTALL_DIR/greenplum_path.sh
 
-	cd "\${1}/gpdb_src/gpAux"
-	source gpdemo/gpdemo-env.sh
+cd "\${GPDB_SRC_DIR}/gpAux"
+source gpdemo/gpdemo-env.sh
 
-	if [ "$overwrite_gpcloud" = "true" ]
```
Comment (adam8157) [inline concourse/scripts/regression_tests_gpcloud.bash]: This is used by `gpcloud_pipeline.yml`, please keep it.

**Comment by adam8157** [comment]
`gpcloud_pipeline.yml` could be abandoned if gpcloud is not under active development (seems so now), and the main pipeline has tests for gpcloud's unit tests and gpcheckcloud tests.

**Comment by larham** [comment]
@adam8157 thanks for the feedback!  We will duplicate the script with a new name, so that your copy for gpcloud_pipeline.yml can remain as you like.

BTW, checking in gpcloud_pipeline.yml  seems like checking in a development tool, to me.  In our common practice, we often fly a temporary pipeline with the "kickoff" script in the repo "continuous_integration", and that tool pauses all jobs except a few. Then a developer can unpause only the jobs of interest, and a developer also temporarily change triggering to be fast. Thus, temporary pipelines for a particular development task are made easy, and those temporary pipelines are not pushed to master, in our common practice.

**Comment by adam8157** [comment]
That `kickoff` script is cool, I suggest this PR adding the other two tests in gpcloud_pipeline.yml into gpcloud installcheck, and retire the gpcloud_pipeline.yml.

How does that sound? Feel free to choose :)

**Comment by goutamtadi1** [comment]
@adam8157 Yes, We would take that up in a separate PR. See email to track that work for safe removing `gpcloud_pipeline.yml `
